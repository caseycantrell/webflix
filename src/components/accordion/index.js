import { useState } from "react";

export default function Accordion({ children, ...restProps }){
    return (
        <Container {...restProps}>
            <Inner>
                {children}
            </Inner>
        </Container>
    )
};

Accordion.Title = function AccordionTitle({ children, ...restProps }) {
    return <Title {...restProps}></Title>
};

Accordion.Frame = function AccordionFrame({ children, ...restProps }) {
    return <Frame {...restProps}></Frame>
};

Accordion.Item = function AccordionItem({ children, ...restProps }) {
    const [toggleShow, setToggleShow] = useState(false);
    return <Item {...restProps}></Item>
};

Accordion.Header = function AccordionHeader({ children, ...restProps }) {
    return <Header {...restProps}></Header>
};