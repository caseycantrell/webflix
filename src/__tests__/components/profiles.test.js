import { getByTestId, getByText, render } from "@testing-library/react";
import { Profiles } from "../../components";

describe("<Profiles />", () => {
    it("Renders the <Profiles /> component with data", () => {
        const { container, getByText, getByTestId } = render(
        <Profiles>
            <Profiles.Title>Who's watching?</Profiles.Title>
            <Profiles.List>
                <Profiles.Item onClick={() => {}}>
                    <Profiles.Picture src="/images/casey.png" data-testid="profile-picture" />
                    <Profiles.Name>Casey</Profiles.Name>
                </Profiles.Item>
            </Profiles.List>
        </Profiles>
        );
        
        expect(getByText("Who's watching?")).toBeTruthy();
        expect(getByTestId("profile-picture")).toBeTruthy();
        expect(getByText("Casey")).toBeTruthy();
        expect(container.firstChild).toMatchSnapshot();

    });

    it("renders the <Profiles /> component with data and misc profile img", () => {
        const { container, getByText, getByTestId } = render(
        <Profiles>
            <Profiles.Title>Who's watching?</Profiles.Title>
            <Profiles.List>
                <Profiles.Item onClick={() => {}}>
                    <Profiles.Picture src="/images/casey.png" data-testid="profile-picture-misc" />
                    <Profiles.Name>Casey</Profiles.Name>
                </Profiles.Item>
            </Profiles.List>
        </Profiles>
        );

        expect(getByText("Who's watching?")).toBeTruthy();
        expect(getByTestId("profile-picture-misc")).toBeTruthy();
        expect(getByText("Casey")).toBeTruthy();
        expect(container.firstChild).toMatchSnapshot();

    });
});