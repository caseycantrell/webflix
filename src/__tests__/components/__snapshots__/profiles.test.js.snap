// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profiles /> Renders the <Profiles /> component with data 1`] = `
<div
  class="sc-ehCJOs cFmhFO"
>
  <h1
    class="sc-lcepkR erHrWf"
  >
    Who's watching?
  </h1>
  <ul
    class="sc-gGCDDS cSVQTS"
  >
    <li
      class="sc-Galmp javCcL"
    >
      <img
        class="sc-faUpoM dUFagj"
        data-testid="profile-picture"
        src="/images/users//images/casey.png.png"
      />
      <p
        class="sc-clIzBv eWFiLO"
      >
        Casey
      </p>
    </li>
  </ul>
</div>
`;

exports[`<Profiles /> renders the <Profiles /> component with data and misc profile img 1`] = `
<div
  class="sc-ehCJOs cFmhFO"
>
  <h1
    class="sc-lcepkR erHrWf"
  >
    Who's watching?
  </h1>
  <ul
    class="sc-gGCDDS cSVQTS"
  >
    <li
      class="sc-Galmp javCcL"
    >
      <img
        class="sc-faUpoM dUFagj"
        data-testid="profile-picture-misc"
        src="/images/users//images/casey.png.png"
      />
      <p
        class="sc-clIzBv eWFiLO"
      >
        Casey
      </p>
    </li>
  </ul>
</div>
`;
